### DISCUSSION TITLE
import / export functionality

### DISCUSSION THREAD

**Discussion by ddbit** [chaosprint/glicol#117] [Q&A]
Hi, this is a great project. I've just learned the basics and actually enjoying it a lot. However, I immediately found a friction to create some complex pattern and I urge to create some components reusable from a tune to another. That's what devs call a library. I checked the docs but found nothing like a library import/export. Is it a missing feature or I have to search deeper? Any hints?

**Comment by chaosprint** [comment]
That's an interesting point and this feature is currently missing in Glicol. I thought about it before and considered providing some C-style macro, but then I decided not to add that at least for now.

First, I was worried that it may overcomplicate the syntax. I already found it quite challenging to introduce different concepts to beginners who have no programming background, and macro can be even more challenging. 

Another reason not to have it is that in music live coding, it's better to show/share things. Yet importing something may be against this philosophy (by using `import` you hide some codes that could have been shown on the screen). As I like to improve new stuff every time, I wasn't so sure which part needs reusing. For those necessary parts/libs that may be useful to everyone, like the reverb, they are written in Rust and precompiled.

Would you mind posting some examples of the parts you want to reuse?

**Comment by ddbit** [comment]
Hi, I probably misunderstood the goal of this tool which seems a fit for
adjusting and tweaking code during live sessions and not for creating long
tunes based on some libraries. It makes a new sense to me now. I see your
point about overcomplicating things and I think you have a case.
I'm just getting started and have no real stuff to show atm.

On another subject I wonder whether it is possible (or will be) extending
the samples library, but will open another topic for that.



On Mon, Oct 10, 2022 at 8:38 PM Qichao Lan ***@***.***> wrote:

> That's an interesting point and this feature is currently missing in
> Glicol. I thought about it before and considered providing some C-style
> macro, but then I decided not to add that at least for now.
>
> First, I was worried that it may overcomplicate the syntax. I already
> found it quite challenging to introduce different concepts to beginners who
> have no programming background, and macro can be even more challenging.
>
> Another reason not to have it is that in music live coding, it's better to
> show/share things. Yet importing something may be against this philosophy
> (by using import you hide some codes that could have been shown on the
> screen). As I like to improve new stuff every time, I wasn't so sure which
> part needs reusing. For those necessary parts/libs that may be useful to
> everyone, like the reverb, they are written in Rust and precompiled.
>
> Would you mind posting some examples of the parts you want to reuse?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/chaosprint/glicol/discussions/117#discussioncomment-3841951>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIC4IUBZFC2SD2MFHNJXWITWCRPCZANCNFSM6AAAAAARBCB5LQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 



*Ing. Davide Carboni, PhD Partita IVA 03800580924 -- **Albo sez. A (CA) n.
7236*


*Viale La Plaia 15 - 09123 CagliariTel: +39 3293547783 *
*Cod. Destinatario SDI: *M5UXCR1
PEC: ***@***.*** ***@***.***>*
Book call on Calendly <https://calendly.com/dcarboni-1/30min>
Blog -- http://digitaldavide.me
